### ISSUE TITLE
Add support for file-like objects to mne.io.read_raw_edf()

### ISSUE THREAD

**Issue by CJ-Wave** [mne-tools/mne-python#8976]
#### Describe the new feature or enhancement
I would like to have support for file-like objects for the mne.io.read_raw_edf() function (specifically a bytesIO object)

#### Describe your proposed implementation
new capability to an existing method

#### Describe possible alternatives
Possible alternative is allowing for a kwarg to detect if it's a file-like object as opposed to a filepath, but shouldn't be necessary

#### Additional comments
It looks like fiff files have that functionality in mne.io.read_raw(), it would be fantastic if edf had that functionality as well

**Comment by larsoner** [comment]
Agreed this would be useful and I don't see any reason offhand why it would be a problem. Maybe some code in `mne/io/fiff/raw.py` could be refactored for this. Do you want to give it a try @CJ-Wave ?

**Comment by CJ-Wave** [comment]
Sure, I can give it a try.

**Comment by pb2640** [comment]
> Sure, I can give it a try.
Hi @CJ-Wave, I was wondering if you were able to implement this feature?

**Comment by szz-dvl** [comment]
Hi, I'd love this functionality too, is there any plan to implement it soon? may I help somehow?

Thanks

**Comment by larsoner** [comment]
@szz-dvl I don't tihnk anyone else is working on it, help is welcome!

**Comment by szz-dvl** [comment]
@larsoner I'm trying to test my changes for GDF files, I downloaded test data from [here](https://github.com/mne-tools/mne-testing-data/tree/master), most of the tests are failing with `io.UnsupportedOperation: fileno`, the error comes from the calls to `np.fromfile` in the function `_read_gdf_header`. By changing `np.fromfile` to `np.frombuffer` and adapting the parameters I can get rid of the error, but, I'm wondering if somebody can check this in another environment before performing this changes, I'm not sure if it is a numpy version issue, I'm using numpy 2.2.3.

Thanks!

**Comment by larsoner** [comment]
@szz-dvl I would expect a change like that to be needed, yeah. The easiest thing to do is probably to open a pull request with your changes. The CIs will then run tests on all OSes and we'll see if it works and what (if anything) breaks!

**Comment by szz-dvl** [comment]
@larsoner Please forget about np.frombuffer, it was on my side.

Forgive any inconvenience.
